File: src/main/java/thirtytwo/degrees/HelloController.java
package thirtytwo.degrees;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

import java.io.IOException;
import java.io.Writer;

@Controller
public class HelloController {

    @RequestMapping(value = "/hello")
    public void home(Writer out) throws IOException {
        System.out.println("HomeController: Passing through...");
        out.write("hello heroku");
        out.flush();
        out.close();
    }
}

File: scala_2.9.1/src/main/scala/thirtytwo/degrees/halfpipe/scalaexample/view/HelloController.scala
package thirtytwo.degrees.halfpipe.scalaexample.view

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import java.io.Writer

/**
 * User: spencergibb
 * Date: 9/22/12
 * Time: 4:19 PM
 */
@Controller
class HelloController {

  @RequestMapping(Array("/hello"))
  def hello(out: Writer) = {
    println("HelloController: Passing through...")
    out.write("hello controller")
    out.flush()
    out.close()
  }
}
